source/common/router/config_impl.cc

```diff
@@ -1404,10 +1404,11 @@ absl::optional<std::string> PathTemplateRouteEntryImpl::currentUrlPathAfterRewri
   return currentUrlPathAfterRewriteWithMatchedPath(headers, "");
 }
 
-RouteConstSharedPtr PathTemplateRouteEntryImpl::matches(const Http::RequestHeaderMap& headers,
-                                                        const StreamInfo::StreamInfo& stream_info,
-                                                        uint64_t random_value) const {
+RouteConstSharedPtr PathTemplateRouteEntryImpl::matches(
+    const Http::RequestHeaderMap& headers, const StreamInfo::StreamInfo& stream_info,
+    uint64_t random_value, bool ignore_path_parameters_in_path_matching) const {
   throw absl::UnimplementedError("Path template matcher not implemented");
+  (void)ignore_path_parameters_in_path_matching;
```

Comment:
void a bool value?

Reply:
void silences compiler unused warning. I will comment on it.

Reply:
What about using an anonymous parameter?

Reply:
Probably an UNREFERENCED_PARAMETER wrapper on the variable?  WDYT? https://github.com/envoyproxy/envoy/blob/0b10b055a6e1dbdd600f9a80f8b8d803bdd8ec9c/source/common/common/macros.h#L21-L26

Reply:
You can just simply omit the unused variable name in the function signature to suppress the warning/error.

Reply:
Sounds good.